### ISSUE TITLE
oxc-parser: HTML-style line comment value includes marker leftovers

### ISSUE THREAD

**Issue by arshad-yaseen** [oxc-project/oxc#22803]
When parsing Annex B HTML-style line comments (`<!--` and `-->`), the resulting comment `value` contains leftover marker characters. The `start`/`end` ranges are correct, only `value` is wrong.

It looks like `value` is always computed by stripping a fixed **2** leading characters (as for `//`), regardless of the actual marker length: `<!--` is 4 chars and `-->` is 3 chars.

**Version:** `oxc-parser@0.133.0` (npm)

**Reproduction:**

```ts
import { parseSync } from "oxc-parser";

for (const src of ["<!--a\n", "-->", "<!--"]) {
  const { comments } = parseSync("test.js", src, { sourceType: "script", lang: "js" })
  console.log(JSON.stringify(src), "=>", comments)
}
```

**Actual:**

```
"<!--a\n" => [{ type: "Line", value: "--a", start: 0, end: 5 }]
"-->"     => [{ type: "Line", value: ">",   start: 0, end: 3 }]
"<!--"    => [{ type: "Line", value: "--",  start: 0, end: 4 }]
```

**Expected** (text after the full marker):

```
"<!--a\n" => [{ type: "Line", value: "a", start: 0, end: 5 }]
"-->"     => [{ type: "Line", value: "",  start: 0, end: 3 }]
"<!--"    => [{ type: "Line", value: "",  start: 0, end: 4 }]
```

Both `@babel/parser@7.29.3` and `acorn@8.16.0` produce `"a"`, `""`, `""` for these inputs. (`@typescript-eslint/parser` doesn't implement Annex B HTML comments and reports a parse error instead, so it's not a `value` reference.)

**Comment by babu-ch** [comment]
The leftover characters come from `value` stripping a fixed **2** leading chars — fine for `//`, wrong for `<!--` (4) and `-->` (3).

Root cause: `Comment::content_span()` in `oxc_ast` assumes a 2-char line delimiter, and both value paths (NAPI + raw transfer) inherit it.

Two ways to fix:
- **Non-breaking** — fix only the value computation; `content_span()` stays as-is (so it remains wrong for HTML-style comments used elsewhere).
- **Breaking** — add a `source_text` param to `content_span()` and fix it at the source; changes a public `oxc_ast` API.

Which approach would the team prefer?
